plenum/test/checkpoints/test_backup_replica_resumes_ordering_on_lag_in_checkpoints.py

```diff
@@ -0,0 +1,211 @@
+def test_backup_replica_resumes_ordering_on_lag_in_checkpoints(
+        looper, chkFreqPatched, reqs_for_checkpoint,
+        one_replica_and_others_in_backup_instance,
+        sdk_pool_handle, sdk_wallet_client, view_change_done):
+    """
+    Verifies resumption of ordering 3PC-batches on a backup replica
+    on detection of a lag in checkpoints
+    """
+
+    slow_replica, other_replicas = one_replica_and_others_in_backup_instance
+    view_no = slow_replica.viewNo
+
+    # Send a request and ensure that the replica orders the batch for it
+    sdk_send_random_requests(looper, sdk_pool_handle, sdk_wallet_client, 1)
+
+    looper.run(
+        eventually(lambda: assertExp(slow_replica.last_ordered_3pc == (view_no, 1)),
+                   retryWait=1,
+                   timeout=waits.expectedTransactionExecutionTime(4)))
+
+    # Don't receive Commits from two replicas
+    slow_replica.node.nodeIbStasher.delay(
+        cDelay(instId=1, sender_filter=other_replicas[0].node.name))
+    slow_replica.node.nodeIbStasher.delay(
+        cDelay(instId=1, sender_filter=other_replicas[1].node.name))
+
+    # Send a request for which the replica will not be able to order the batch
+    # due to an insufficient count of Commits
+    sdk_send_random_requests(looper, sdk_pool_handle, sdk_wallet_client, 1)
+    looper.runFor(waits.expectedTransactionExecutionTime(4))
+
+    # Recover reception of Commits
+    slow_replica.node.nodeIbStasher.drop_delayeds()
+    slow_replica.node.nodeIbStasher.resetDelays()
+
+    # Send requests but in a quantity insufficient
+    # for catch-up number of checkpoints
+    sdk_send_random_requests(looper, sdk_pool_handle, sdk_wallet_client,
+                             Replica.STASHED_CHECKPOINTS_BEFORE_CATCHUP *
+                             reqs_for_checkpoint - 2)
+    looper.runFor(waits.expectedTransactionExecutionTime(4))
+
+    # Ensure that the replica has not ordered any batches
+    # after the very first one
+    assert slow_replica.last_ordered_3pc == (view_no, 1)
+
+    # Ensure that the watermarks have not been shifted since the view start
+    assert slow_replica.h == 0
+    assert slow_replica.H == 15
+
+    # Ensure that the collections related to requests, batches and
+    # own checkpoints are not empty.
+    # (Note that a primary replica removes requests from requestQueues
+    # when creating a batch with them.)
+    if slow_replica.isPrimary:
+        assert slow_replica.sentPrePrepares
+    else:
+        assert slow_replica.requestQueues[DOMAIN_LEDGER_ID]
+        assert slow_replica.prePrepares
+    assert slow_replica.prepares
+    assert slow_replica.commits
+    assert slow_replica.batches
+    assert slow_replica.checkpoints
+
+    # Ensure that there are some quorumed stashed checkpoints
+    assert slow_replica.stashed_checkpoints_with_quorum()
+
+    # Send more requests to reach catch-up number of checkpoints
+    sdk_send_random_requests(looper, sdk_pool_handle, sdk_wallet_client,
```

Comment:
As I understand, it works since reqs_for_checkpoint =2. Can we send `Replica.STASHED_CHECKPOINTS_BEFORE_CATCHUP * reqs_for_checkpoint - 1` in previous sending, and send `1` here?